wee_slack.py

```diff
@@ -3693,6 +3727,33 @@ def command_status(data, current_buffer, args):
         s = SlackRequest(team.token, "users.profile.set", {"profile": profile}, team_hash=team.team_hash)
         EVENTROUTER.receive(s)
 
+@utf8_decode
+def line_event_cb(data, signal, hashtable):
+    line_pointer = hashtable["_chat_line"]
+    buffer_pointer = hashtable["_buffer"]
+    line_timestamp = hashtable["_chat_line_date"]
+    line_time_id = hashtable["_chat_line_date_printed"]
+
+    e = EVENTROUTER
+    if line_timestamp and line_time_id and buffer_pointer in e.weechat_controller.buffers:
+        ts = SlackTS("{}.{}".format(line_timestamp, line_time_id))
+
+        channel = e.weechat_controller.buffers[buffer_pointer]
+        message_hash = channel.hash_message(ts)
```

Comment:
This fails with a stacktrace if you try to use cursor mode in a team buffer.

Reply:
Fixed

Reply:
Just a small improvement, this can be simplified by doing the check in the first if, instead of a nested if with a separate return.

```diff
diff --git a/wee_slack.py b/wee_slack.py
index fcbc293..d1a526b 100644
--- a/wee_slack.py
+++ b/wee_slack.py
@@ -3732,14 +3732,11 @@ def line_event_cb(data, signal, hashtable):
     buffer_pointer = hashtable["_buffer"]
     line_timestamp = hashtable["_chat_line_date"]
     line_time_id = hashtable["_chat_line_date_printed"]
+    channel = EVENTROUTER.weechat_controller.buffers.get(buffer_pointer)
 
-    e = EVENTROUTER
-    if line_timestamp and line_time_id and buffer_pointer in e.weechat_controller.buffers:
+    if line_timestamp and line_time_id and isinstance(channel, SlackChannelCommon):
         ts = SlackTS("{}.{}".format(line_timestamp, line_time_id))
 
-        channel = e.weechat_controller.buffers[buffer_pointer]
-        if not isinstance(channel, SlackChannelCommon):
-            return w.WEECHAT_RC_OK
         message_hash = channel.hash_message(ts)
         if message_hash is None:
             return w.WEECHAT_RC_OK
```

Reply:
:+1: